mailserver/mailserver.go

```diff
@@ -286,14 +303,20 @@ func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bl
 					log.Error(fmt.Sprintf("Failed to send direct message to peer: %s", err))
 					return
 				}
+				lastEnvelopeHash = envelope.Hash()
 			}
 			sentEnvelopes++
 			sentEnvelopesSize += whisper.EnvelopeHeaderLength + int64(len(envelope.Data))
+
+			if limit != 0 && sentEnvelopes == limit {
```

Comment:
`limit == 0` should be checked earlier. If it's `0` then we still send one message.

Reply:
since it's an integer, 0 is the default so I manage 0 as if there's no limit, to keep compatibility with old clients without pagination feature. does it make sense?

Reply:
Ahhh I see. Yeah it does but we should add an explanation for that. A comment or `const NoLimit = 0` and compare `if limit != NoLimit`.